nova/virt/vmwareapi/vmops.py

```diff
@@ -1552,6 +1558,49 @@ def _clean_up_after_special_spawning(self, context, instance_memory_mb,
                 placement_client.set_inventory_for_provider(context, rp.uuid,
                                                             inv_data)
 
+    def _stack_vm_to_host_if_needed(self, instance):
+        """Selects the fullest host that can fit the VM.
+
+        This runs only if DRS is disabled and the configuration
+        CONF.vmware.drs_disabled_stack_vms is set to True
+
+        :returns: HostSystem mo-ref or None
+        """
+        if not CONF.vmware.drs_disabled_stack_vms:
+            return None
+        if cluster_util.is_drs_enabled(self._session, self._cluster):
+            return None
+        LOG.info("DRS is disabled and stacking VMs was enabled.")
+
+        hosts_free = []
+        hosts_full = []
+        for host_ref, host in vm_util.get_stats_from_cluster_per_host(
+                self._session, self._cluster).items():
+            memory_mb_free = (host['memory_mb'] -
```

Comment:
Do we have to filter for `available` or are we only getting hosts that can accept VMs (not disconnected, not failover host)?